creator/html_paragraph.go

```diff
@@ -0,0 +1,625 @@
+				t.rows = append(t.rows, &htmlTableRow{})
+			}
+		case "td", "th":
+			if t := b.tableStack.currentTable(); t != nil && len(t.rows) > 0 {
+				newB = newHTMLBlock(b, b.styleStack.currentStyle())
+				style := newB.parseNodeStyle(node)
+				newB.style = style
+				newB.styleStack.pushStyle(style)
+				defer newB.styleStack.popStyle()
+
+				row := t.rows[len(t.rows)-1]
+				cell := htmlTableCell{block: newB}
+				row.cells = append(row.cells, &cell)
+				if l := len(row.cells); l > t.maxColIndex {
+					t.maxColIndex = l
+				}
+			}
+			if node.Data == "th" {
+				newB.styleStack.pushStyle(newB.styleStack.addBold())
+				defer newB.styleStack.popStyle()
+			}
+		case "div":
+			newB = newHTMLBlock(b, b.styleStack.currentStyle())
+			b.elements = append(b.elements, newB)
+			b.currentParagraph = nil
+			style := newB.parseNodeStyle(node)
+			newB.style = style
+			newB.styleStack.pushStyle(style)
+			defer newB.styleStack.popStyle()
+		case "p":
+			style := b.parseNodeStyle(node)
+			b.styleStack.pushStyle(style)
+			defer b.styleStack.popStyle()
+			b.currentParagraph = newStyledParagraph(b.styleStack.currentStyle().TextStyle)
+			b.currentParagraph.alignment = b.styleStack.currentStyle().TextAlignment
+			b.elements = append(b.elements, b.currentParagraph)
+		case "br":
+			//b.currentParagraph = newStyledParagraph(b.styleStack.currentStyle().TextStyle)
+			//b.currentParagraph.alignment = b.styleStack.currentStyle().TextAlignment
+			//b.elements = append(b.elements, b.currentParagraph)
+			//b.currentParagraph.Append("\n")
+			p, created := b.getCurrentOrCreateParagraph()
+			if created {
+				b.currentParagraph.alignment = b.styleStack.currentStyle().TextAlignment
+			} else {
+				p.Append("\n")
+			}
+		case "b":
+			b.styleStack.pushStyle(b.styleStack.addBold())
+			defer b.styleStack.popStyle()
+		case "i":
+			b.styleStack.pushStyle(b.styleStack.addItalic())
+			defer b.styleStack.popStyle()
+		}
+	}
+
+	for next := node.FirstChild; next != nil; next = next.NextSibling {
+		if err := newB.processNode(next); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// HTMLParagraph  allow to use simple HTML markup to generate content.
+type HTMLParagraph struct {
+	blocks     []*htmlBlock
+	tableStack htmlTableStack
+	styleStack htmlStyleStack
+}
+
+func (b *htmlBlock) getCurrentOrCreateParagraph() (*StyledParagraph, bool) {
+	if b.currentParagraph == nil {
+		b.currentParagraph = newStyledParagraph(b.styleStack.RegularStyle)
+		b.elements = append(b.elements, b.currentParagraph)
+		return b.currentParagraph, true
+	}
+	return b.currentParagraph, false
+}
+
+// Width returns the width of the Drawable.
+func (b *htmlBlock) Width() float64 {
+	if es := b.style.ElementStyle; es != nil && es.width != nil {
+		return *es.width
+	}
+	var w float64
+	for _, e := range b.elements {
+		w = math.Max(w, e.Width())
+	}
+	return w
+}
+
+// Height returns the height of the Drawable.
+func (b *htmlBlock) Height() float64 {
```

Comment:
Shouldn't the total height be the sum of the element heights?

Reply:
Thanks, I fixed it.